lib/opencast-codebuild-stack.ts

```diff
@@ -63,62 +63,83 @@ export class OpencastCodebuild extends cdk.Stack {
       removalPolicy: cdk.RemovalPolicy.DESTROY,
     });
 
-    const computeEnvironment: codebuild.BuildEnvironment = {
-      computeType: codebuild.ComputeType.LARGE,
-      buildImage: codebuild.LinuxBuildImage.fromEcrRepository(
-        ecr.Repository.fromRepositoryName(this, 'EcrRepository', 'hdce/oc-codebuild-environment'),
-        '1.0.1',
-      ),
-    };
-
     const environmentVariables = {
       NOTIFY_FUNCTION: { value: notifyFunction.functionArn },
     }
 
-    const artifacts: codebuild.IArtifacts = codebuild.Artifacts.s3({
-      bucket: buildBucket,
-      includeBuildId: false,
-      packageZip: false,
-      encryption: false,
-    });
-
     // this depends on the oauth connection to bitbucket being established already; done via web console
     const opencastSource = {
-      webhook: true,
       owner: 'hudcede',
       repo: 'matterhorn-dce-fork',
+      cloneDepth: 1,
+      report: false,
     };
 
-    const cookbookSource = {
-      webhook: true,
+    const imageSource = {
       owner: 'harvard-dce',
-      repo: 'mh-opsworks-recipes',
+      repo: 'opencast-ecs-images',
+      cloneDepth: 1,
+      report: false,
     };
 
-    const buildProject = new codebuild.Project(this, 'BuildProject', {
-      projectName: `${cdkStackName}-build`,
-      artifacts,
-      environmentVariables,
-      environment: computeEnvironment,
+    const imageBuildProject = new codebuild.Project(this, 'BuildImageProject', {
+      projectName: `${cdkStackName}-image-build`,
+      environmentVariables: {
+        ...environmentVariables,
+        IMAGE_REGISTRY: {
+          value: `${cdk.Stack.of(this).account}.dkr.ecr.us-east-1.amazonaws.com/hdce`,
+          type: codebuild.BuildEnvironmentVariableType.PLAINTEXT,
+        },
+        OPENCAST_REPO: {
```

Comment:
Should this be added to the parameter store config instead?

Reply:
No this is the correct way to do it IMO. Putting it in the parameter store config would mean it has to go into the codebuild environment variables in plain text, right? That we don't want.